src/Symfony/Component/Workflow/Marking.php

```diff
@@ -16,35 +16,93 @@
  *
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  */
+/**
+ * Class Marking
+ * @package Symfony\Component\Workflow
+ */
 class Marking
 {
+    /**
+     * @var string[]
+     */
     private $places = array();
 
     /**
-     * @param array $places keys are the place name and values should be 1
+     * @param string[] $places keys are the place name and values should be 1
      */
     public function __construct(array $places = array())
     {
         foreach ($places as $place => $nbToken) {
-            $this->mark($place);
+            $this->markPlaceNamed($place);
         }
     }
 
-    public function mark($place)
+    /**
+     * @param Place[] $places
+     * @return Marking
+     */
+    public static function fromPlaces(array $places = array())
```

Comment:
I removed this method because it's not the responsibility of this class to do that.